### PR TITLE
chore: setup repo

### PR DESCRIPTION
**Pull request by mds1**
Adds a basic README, and configures a few solidity options and slither settings

### CODE REVIEW

**Comment by mds1** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/ScopeLift/stealth-address-erc-contracts/1" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/stealth-address-erc-contracts/1?utm_source=stack-comment).

File: foundry.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [profile.default]
   optimizer = true
   optimizer_runs = 10_000_000
-  solc_version = "0.8.16"
+  solc_version = "0.8.20"
```
Comment (alexkeating) [inline foundry.toml:5]: Are we assuming by the time this is released L2's would have implemented `PUSH0`? Or is this a non-issue

**Review by alexkeating — APPROVED** [review]
Looks good, left a small question on the version

File: foundry.toml (hunk 1 above)
Comment (mds1) [reply foundry.toml:5]: See https://github.com/ScopeLift/foundry-template/pull/20#discussion_r1256017884, but pinned EVM version in e0c9e78

**Review by jferas — APPROVED** [review]

**Pull request merged**